sql/catalyst/src/main/scala/org/apache/spark/sql/internal/connector/SimpleTableProvider.scala

```diff
@@ -45,7 +45,6 @@ trait SimpleTableProvider extends TableProvider {
       schema: StructType,
       partitioning: Array[Transform],
       properties: util.Map[String, String]): Table = {
-    assert(partitioning.isEmpty)
```

Comment:
Do you know the reason about the assert? If not, please revert it.

Reply:
+1